lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -151,56 +158,48 @@ def execute(module):
 
 
 def update_acl(module):
-
     rules = module.params.get('rules')
-    state = module.params.get('state')
     token = module.params.get('token')
     token_type = module.params.get('token_type')
     mgmt = module.params.get('mgmt_token')
     name = module.params.get('name')
     consul = get_consul_api(module, mgmt)
     changed = False
 
-    try:
+    rules = decode_rules_as_yml(rules)
+    rules_as_hcl = encode_rules_as_hcl_string(rules) if len(rules) > 0 else None
 
+    try:
         if token:
             existing_rules = load_rules_for_token(module, consul, token)
-            supplied_rules = yml_to_rules(module, rules)
-            changed = not existing_rules == supplied_rules
+            changed = existing_rules != rules
             if changed:
-                y = supplied_rules.to_hcl()
                 token = consul.acl.update(
                     token,
                     name=name,
                     type=token_type,
-                    rules=supplied_rules.to_hcl())
+                    rules=rules_as_hcl)
         else:
             try:
-                rules = yml_to_rules(module, rules)
-                if rules.are_rules():
-                    rules = rules.to_hcl()
-                else:
-                    rules = None
-
-                token = consul.acl.create(
-                    name=name, type=token_type, rules=rules)
+                token = consul.acl.create(name=name, type=token_type, rules=rules_as_hcl)
                 changed = True
             except Exception as e:
                 module.fail_json(
-                    msg="No token returned, check your management key and that \
-                         the host is in the acl datacenter %s" % e)
+                    msg="No token returned, check your management key and that"
+                        "the host is in the acl datacenter %s" % e)
     except Exception as e:
         module.fail_json(msg="Could not create/update acl %s" % e)
 
+    # TODO: It is unclear to me why `token`, `rules`, `name` and `type` are returned - they are all inputs. The only one
+    # I can see being useful is the JSON view of the rules.
```

Comment:
Good point - I overlooked that the token can be created. I suggest that we take out `name` and `token_type`, given that they're not likely to be useful in the return.

I would also like to float the idea that we return the remaining rules when an ACL is removed. Although this would require an extra API call, I can see it being helpful (particularly when debugging) and it would make the return consistent.

I'll complete the `RETURN` documentation once we come to a conclusion.